capture/src/capture.cpp

```diff
@@ -58,9 +58,17 @@ static void set_thread_priority(pthread_t thread, int policy, int priority)
 {
     sched_param sch_params;
     sch_params.sched_priority = priority;
-    if (pthread_setschedparam(thread, policy, &sch_params))
+    if ((errno = pthread_setschedparam(thread, policy, &sch_params)))
```

Comment:
Oh hmm. I was under the impression that system calls returned -1 on error, and set `errno` automatically. Why does this break that convention?

Reply:
I don't necessarily know the "why" behind it, to be honest.

All I know is that pthread does seem to have its own self-consistent convention where it generally returns an errno value directly, without setting the `errno` global variable *per se*.

The "did this call succeed" check is similar-ish to standard Unix calls (see if return value was nonzero). The main difference is just that if you're gonna do something afterwards which presupposes that `errno` was actually set by the call (such as use the `err(3)` family of BSD printing functions), you kinda have to shim it by assigning `errno` yourself. (And AFAIK it's not a particularly heinous crime to assign to `errno` manually.)